lib/mongo/client.rb

```diff
@@ -814,6 +808,41 @@ def watch(pipeline = [], options = {})
 
     private
 
+    # Generate default client options based on the URI and options
+    # passed into the Client constructor.
+    def default_options(options)
+      Database::DEFAULT_OPTIONS.dup.tap do |default_options|
+        if options[:auth_mech] || options[:user]
+          default_options[:auth_source] = default_auth_source(options)
+        end
+
+        if options[:auth_mech] == :gssapi
+          opts[:auth_mech_properties] = { service_name: 'mongodb' }
+        end
+
+        # Since we know that the only URI option that sets :ssl_cert is "tlsCertificateKeyFile", any
+        # value set for :ssl_cert must also be set for :ssl_key.
+        if options[:ssl_cert]
+          default_options[:ssl_key] = options[:ssl_cert]
```

Comment:
This expansion makes sense for URI processing because one URI option maps to two Ruby options. The Ruby options however, as documented in https://docs.mongodb.com/ruby-driver/master/tutorials/ruby-driver-create-client/#ruby-options, are not the same. In particular, ssl_key is not documented to accept a file containing both a key and a certificate; if this is the case, can ssl_key documentation be updated to state this?

Furthermore, the ssl_key documentation currently states:

> Note that even if the key is stored in the same file as the certificate, both need to be explicitly specified. 

Thus we are apparently already requiring a user to provide both ssl_key and ssl_cert options explicitly.

Reply:
Thanks for all this. I'll update it!